exchanges/exchange.go

```diff
@@ -1943,3 +1946,57 @@ func (b *Base) Bootstrap(_ context.Context) (continueBootstrap bool, err error)
 func (b *Base) IsVerbose() bool {
 	return b.Verbose
 }
+
+// GetDefaultConfig returns a default exchange config
+func (b *Base) GetDefaultConfig(ctx context.Context, instance LimitedScope) (*config.Exchange, error) {
+	if instance == nil {
+		return nil, errExchangeIsNil
+	}
+
+	if instance.GetBase() != b {
+		return nil, errExchangeMismatch
+	}
+
+	if instance.GetName() == "" {
+		instance.SetDefaults()
+	}
+
+	exchCfg, err := b.GetStandardConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	err = b.SetupDefaults(exchCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	if b.Features.Supports.RESTCapabilities.AutoPairUpdates {
+		err = b.UpdateTradablePairs(ctx, instance)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return exchCfg, nil
+}
+
+// UpdateTradablePairs updates the exchanges available pairs and stores them in
+// the exchanges config.
+func (b *Base) UpdateTradablePairs(ctx context.Context, instance LimitedScope) error {
```

Comment:
One suggestion in both cases in `SetDefaults` we set those calling functions as a base field and call them. Then I can get rid of the jank because it is quite gross.

Reply:
Wait I think that might defeat the purpose. Let me check.